src/main/kotlin/org/rust/ide/inspections/ApproxConstantInspection.kt

```diff
@@ -3,39 +3,30 @@ package org.rust.ide.inspections
 import com.intellij.codeInspection.ProblemsHolder
 import com.intellij.psi.PsiElement
 import com.intellij.psi.PsiElementVisitor
-import org.rust.lang.core.psi.RustLitExpr
-import org.rust.lang.core.psi.RustVisitor
+import com.intellij.psi.util.PsiTreeUtil
+import org.rust.lang.core.psi.RustAttr
+import org.rust.lang.core.psi.RustLiteral
+import org.rust.lang.core.psi.RustNumber
+import org.rust.lang.core.psi.RustTokenElementTypes
+import org.rust.lang.core.psi.visitors.RustVisitorEx
 
 class ApproxConstantInspection : RustLocalInspectionTool() {
-
     override fun getDisplayName() = "Approximate Constants"
 
-    override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean): PsiElementVisitor {
-        return object : RustVisitor() {
-            override fun visitLitExpr(o: RustLitExpr) {
-                analyzeLiteral(o.floatLiteral ?: return, holder)
+    override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean): PsiElementVisitor =
+        object : RustVisitorEx() {
+            override fun visitNumericLiteral(literal: RustLiteral.Number) {
+                if (literal.tokenType == RustTokenElementTypes.FLOAT_LITERAL &&
+                    PsiTreeUtil.getParentOfType(literal, RustAttr::class.java) == null) {
```

Comment:
We have a more kotlinish api for getParentOfType here: https://github.com/intellij-rust/intellij-rust/blob/master/src/main/kotlin/org/rust/lang/core/psi/util/RustPsiExtensions.kt#L18-L27

Reified type parameters in inline functions are neat!